sources/packages/backend/apps/api/src/route-controllers/overaward/_tests_/e2e/overaward.students.controller.getOverawardsByStudent.e2e-spec.ts

```diff
@@ -0,0 +1,98 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  createFakeDisbursementOveraward,
+  createFakeStudentAssessment,
+  createFakeApplication,
+} from "@sims/test-utils";
+import { Repository } from "typeorm";
+import {
+  Application,
+  AssessmentTriggerType,
+  DisbursementOveraward,
+  DisbursementOverawardOriginType,
+  Student,
+  StudentAssessment,
+} from "@sims/sims-db";
+
+import {
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getStudentToken,
+  FakeStudentUsersTypes,
+  getStudentByFakeStudentUserType,
+} from "../../../../testHelpers";
+import { StudentsOverawardAPIOutDTO } from "../../models/overaward.dto";
+
+describe("OverawardStudentsController(e2e)-getOverawardsByStudent", () => {
+  let app: INestApplication;
+  let assessmentRepo: Repository<StudentAssessment>;
+  let applicationRepo: Repository<Application>;
+  let disbursementOverawardRepo: Repository<DisbursementOveraward>;
+  let student: Student;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    assessmentRepo = dataSource.getRepository(StudentAssessment);
+    applicationRepo = dataSource.getRepository(Application);
+    disbursementOverawardRepo = dataSource.getRepository(DisbursementOveraward);
+    student = await getStudentByFakeStudentUserType(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+      dataSource,
+    );
+  });
+
+  it("Should return student overawards", async () => {
+    // Prepare the student assessment to create overaward.
+    const application = await applicationRepo.save(
+      createFakeApplication({
+        student,
+      }),
+    );
+    const studentAssessment = await assessmentRepo.save(
+      createFakeStudentAssessment({ auditUser: student.user, application }),
+    );
+    application.currentAssessment = studentAssessment;
+    await applicationRepo.save(application);
+    // Create an overaward.
+    const reassessmentOveraward = createFakeDisbursementOveraward({ student });
+    reassessmentOveraward.studentAssessment = studentAssessment;
+    reassessmentOveraward.disbursementValueCode = "CSLP";
+    reassessmentOveraward.overawardValue = 500;
+    reassessmentOveraward.originType =
+      DisbursementOverawardOriginType.ReassessmentOveraward;
+    reassessmentOveraward.addedDate = new Date();
+    await disbursementOverawardRepo.save(reassessmentOveraward);
+    const endpoint = `/students/overaward`;
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(
+        await getStudentToken(FakeStudentUsersTypes.FakeStudentUserType1),
+        BEARER_AUTH_TYPE,
+      )
+      .expect(HttpStatus.OK)
+      .then((response) => {
+        expect(response.body.length).toBeGreaterThan(0);
+        const [overaward] = response.body as StudentsOverawardAPIOutDTO[];
+        expect(overaward.dateAdded).toBe(
```

Comment:
mm, In test 1 we are already adding 2 overward for this same student, and this test is to get all overward belonging to this student, so how can we be certain about the result we are expecting, let me know if I am missing anything

Reply:
Maybe as a temporary solution, since the idea is to try to close this PR to invest time on #1835, we can use the below assert that will not rely on the first object in the array and will also consider the date which minimizes the possible concurrency conflict. We can also leave a TODO to review these test cases after #1835 is done.
@ann-aot @andrepestana-aot @guru-aot thoughts?

```ts
    // Act/Assert
    await request(app.getHttpServer())
      .get(endpoint)
      .auth(studentToken, BEARER_AUTH_TYPE)
      .expect(HttpStatus.OK)
      .then((response) => {
        expect(response.body.length).toBeGreaterThan(0);
        const overawards = response.body as StudentsOverawardAPIOutDTO[];
        const createdOveraward = overawards.find(
          (overaward) =>
            overaward.dateAdded.toString() ===
              reassessmentOveraward.addedDate.toISOString() &&
            overaward.overawardOrigin ===
              DisbursementOverawardOriginType.ReassessmentOveraward &&
            overaward.assessmentTriggerType ===
              AssessmentTriggerType.OriginalAssessment &&
            overaward.awardValueCode === "CSLP" &&
            overaward.overawardValue === 500,
        );
        expect(createdOveraward).toBeDefined();
      });
```

Reply:
I am good

Reply:
I think we could use expect.arrayContaining() like in:

```js
expect(programYearList).toEqual(
          expect.arrayContaining([
            expect.objectContaining({
              id: programYear2004.id,
              description:
                "(2004-2005) - Study starting between August 01, 2004 and July 31, 2005",
            }),
         ])
);
```

Reply:
i am okay to do the change for now.